src/library/trackcollection.cpp

```diff
@@ -181,333 +142,213 @@ void TrackCollection::relocateDirectory(QString oldDir, QString newDir) {
     m_trackDao.databaseTracksMoved(std::move(movedIds), QSet<TrackId>());
 }
 
-#ifdef __SQLITE3__
-// from public domain code
-// http://www.archivum.info/qt-interest@trolltech.com/2008-12/00584/Re-%28Qt-interest%29-Qt-Sqlite-UserDefinedFunction.html
-void TrackCollection::installSorting(QSqlDatabase &db) {
-    QVariant v = db.driver()->handle();
-    if (v.isValid() && strcmp(v.typeName(), "sqlite3*") == 0) {
-        // v.data() returns a pointer to the handle
-        sqlite3* handle = *static_cast<sqlite3**>(v.data());
-        if (handle != 0) { // check that it is not NULL
-            int result = sqlite3_create_collation(
-                    handle,
-                    "localeAwareCompare",
-                    SQLITE_UTF16,
-                    NULL,
-                    sqliteLocaleAwareCompare);
-            if (result != SQLITE_OK)
-            qWarning() << "Could not add string collation function: " << result;
-
-            result = sqlite3_create_function(
-                    handle,
-                    "like",
-                    2,
-                    SQLITE_ANY,
-                    NULL,
-                    sqliteLike,
-                    NULL, NULL);
-            if (result != SQLITE_OK)
-            qWarning() << "Could not add like 2 function: " << result;
-
-            result = sqlite3_create_function(
-                    handle,
-                    "like",
-                    3,
-                    SQLITE_UTF8, // No conversion, Data is stored as UTF8
-                    NULL,
-                    sqliteLike,
-                    NULL, NULL);
-            if (result != SQLITE_OK)
-            qWarning() << "Could not add like 3 function: " << result;
-        } else {
-            qWarning() << "Could not get sqlite handle";
-        }
-    } else {
-        qWarning() << "handle variant returned typename " << v.typeName();
+bool TrackCollection::hideTracks(const QList<TrackId>& trackIds) {
+    // Transactional
+    SqlTransaction transaction(database());
+    DEBUG_ASSERT_AND_HANDLE(transaction) {
+        return false;
     }
-}
+    DEBUG_ASSERT_AND_HANDLE(m_trackDao.onHidingTracks(transaction, trackIds)) {
```

Comment:
This and this one below reads a bit as if they were only called in debug mode. 
This was my fault reading it and a bit if  an issue of the AND_HANDLE macro. 
I have no idea to improve this situation without destroying the doku featured from the macro. 
Do you ?